src/libcore/tests/num/int_log.rs

```diff
@@ -0,0 +1,64 @@
+#[test]
+fn log() {
+    assert_eq!(999u32.log(10), Some(2));
+    assert_eq!(1000u32.log(10), Some(3));
+    assert_eq!(555u32.log(13), Some(2));
+    assert_eq!(63u32.log(4), Some(2));
+    assert_eq!(64u32.log(4), Some(3));
+    assert_eq!(10460353203u64.log(3), Some(21));
+    assert_eq!(10460353202u64.log(3), Some(20));
+    assert_eq!(147808829414345923316083210206383297601u128.log(3), Some(80));
+    assert_eq!(147808829414345923316083210206383297600u128.log(3), Some(79));
+    assert_eq!(22528399544939174411840147874772641u128.log(19683), Some(8));
+    assert_eq!(22528399544939174411840147874772631i128.log(19683), Some(7));
+
+    for i in std::i16::MIN..0 {
```

Comment:
Also, now that the numbers have associated constants, `std::i16::MIN` should become `i16::MIN`. (Same multiple times below.)

Reply:
Oh amazing; I didn't realize this was available on nightly already.